src/oc/web/actions.cljs

```diff
@@ -1002,4 +1006,82 @@
+        ;; Refresh the topic data if the action coming in is from the current user
+        ;; to get the new links to interact with
+        (when is-current-user
+          (api/load-entries topic-slug (utils/link-for (:links topic-data) "collection")))
+        ;; Animate the comments count if we don't have already the same number of comments locally
+        (when (not= (count all-old-comments-data) (count new-comments-data))
+          (utils/pulse-comments-count topic-slug entry-uuid))
+        (assoc-in db comments-key sorted-comments-data))
+      ;; the entry is not present, refresh the full topic
+      (do
+        ;; force refresh of topic
+        (api/load-entries topic-slug (utils/link-for (:links topic-data) "collection"))
+        db))))
+
+(defn- update-reaction
+  "Need to update the local state with the data we have, if the interaction is from the actual unchecked-short
+   we need to refresh the entry since we don't have the links to delete/add the reaction."
+  [db interaction-data add-event?]
+  (let [org-slug (router/current-org-slug)
+        board-slug (router/current-board-slug)
+        topic-slug (keyword (:topic interaction-data))
+        entry-uuid (:entry-uuid interaction-data)
+        topic-data (dispatcher/topic-data db org-slug board-slug topic-slug)
+        entry-data (dispatcher/entry entry-uuid)]
+    (if (and entry-data (:reactions entry-data))
+      (let [reaction-data (:interaction interaction-data)
+            old-reactions-data (:reactions entry-data)
+            reaction-idx (utils/index-of old-reactions-data #(= (:reaction %) (:reaction reaction-data)))
+            new-reaction-data {:count (:count interaction-data)}
+            is-current-user (= (jwt/get-key :user-id) (:user-id (:author reaction-data)))
+            with-reacted (if is-current-user
+                            (assoc new-reaction-data :reacted add-event?)
+                            new-reaction-data)
+            new-reactions-data (assoc old-reactions-data reaction-idx (merge (get old-reactions-data reaction-idx) with-reacted))
+            updated-entry-data (assoc entry-data :reactions new-reactions-data)
+            topic-entries-key (dispatcher/topic-entries-key org-slug board-slug topic-slug)
+            entries-data (get-in db topic-entries-key)
+            entry-idx (utils/index-of entries-data #(= (:uuid %) entry-uuid))
+            entry-key (conj topic-entries-key entry-idx)]
+        ;; Refresh the topic data if the action coming in is from the current user
+        ;; to get the new links to interact with
+        (when is-current-user
+          (api/load-entries topic-slug (utils/link-for (:links topic-data) "collection")))
+        (when (not= (:count (get old-reactions-data reaction-idx)) (:count interaction-data))
+          (utils/pulse-reaction-count topic-slug entry-uuid (:reaction reaction-data)))
+        (assoc-in db entry-key updated-entry-data))
```

Comment:
nope, `entry-key` is a vector of keywords and integers https://github.com/open-company/open-company-web/blob/dynamic/src/oc/web/actions.cljs#L1067